src/views/clusteroverview/SettingsTab/ClusterTab/components/ResourceManagementSection/components/KernelSamepageMerging/KernelSamepageMerging.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { FC, useEffect, useState } from 'react';
+
+import HyperConvergedModel from '@kubevirt-ui/kubevirt-api/console/models/HyperConvergedModel';
+import HelpTextIcon from '@kubevirt-utils/components/HelpTextIcon/HelpTextIcon';
+import useHyperConvergeConfiguration, {
+  HyperConverged,
+} from '@kubevirt-utils/hooks/useHyperConvergeConfiguration';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { k8sPatch } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  AlertVariant,
+  PopoverPosition,
+  Split,
+  SplitItem,
+  Switch,
+} from '@patternfly/react-core';
+
+import './KernelSamepageMerging.scss';
+
+const KernelSamepageMerging: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [hyperConverge, hyperLoaded, hyperLoadingError] = useHyperConvergeConfiguration();
+  const [error, setError] = useState<Error>(null);
+
+  useEffect(() => {
+    if (!hyperLoaded || hyperLoadingError) {
+      setError(hyperLoadingError);
+    } else {
+      const ksmConfiguration = hyperConverge?.spec?.configuration?.ksmConfiguration;
+
+      setError(null);
+      setIsEnabled(!!(ksmConfiguration && isEmpty(ksmConfiguration?.nodeLabelSelector))); // Empty nodeLabelSelector will enable KSM on every node.
+    }
+  }, [hyperConverge, hyperLoaded, hyperLoadingError]);
+
+  const onKSMchange = async (value: boolean) => {
+    setIsEnabled(value);
+
+    k8sPatch<HyperConverged>({
+      data: [
+        {
+          op: 'replace',
+          path: `/spec/configuration/ksmConfiguration/nodeLabelSelector`,
+          value: value ? {} : null,
+        },
+      ],
+      model: HyperConvergedModel,
+      resource: hyperConverge,
+    })
+      .then(() => setIsEnabled(value))
```

Comment:
u already doing it on line 40 - why u need here also?

Reply:
but u already doing it on line 40 - u just double doing it without a reason it don't make any sense.
in line 40 u set the enable already no matter if it will pass or failed, so if u call will fail u will have false enabled/disabled. u need to make the logic correct here.

Reply:
I see the point. There is something I forgot to delete, haha. Thanks!